### PR TITLE
NMS-12971: Identify message broker strategies in web "about" page

### PR DESCRIPTION
**Pull request by jeffgdotorg**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12971

### CODE REVIEW

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 1]
```diff
@@ -135,6 +135,10 @@
     <th>Time-Series Strategy:</th>
     <td><%=TimeSeries.getTimeseriesStrategy().getDescr()%></td>
   </tr>
+  <tr>
+    <th>Message Broker Strategy:</th>
+    <td>RPC: <%=System.getProperty("org.opennms.core.ipc.rpc.strategy", "ActiveMQ")%>; Sink: <%=System.getProperty("org.opennms.core.ipc.sink.strategy", "ActiveMQ")%></td>
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:140]: This looks good enough but there is also `GRPC` as IPC strategy for both Sink and RPC which we set through system property `org.opennms.core.ipc.strategy` which takes precedence over  `Sink` and `RPC` specific strategies.

Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/about/index.jsp:140]: > This looks good enough but there is also `GRPC` as IPC strategy for both Sink and RPC which we set through system property `org.opennms.core.ipc.strategy`. This takes precedence over `Sink` and `RPC` specific strategies.

Great catch. Thanks, I'll update the PR to cover this case as well.

**Review by cgorantla — APPROVED** [review]

**Comment by cgorantla** [comment]
@jeffgdotorg Think we should probably try to use existing [method](https://github.com/OpenNMS/opennms/blob/413fdc2b9451ba0c944141cd6b335abfcd8cd936/core/ipc/sink/common/src/main/java/org/opennms/core/ipc/sink/common/SinkStrategy.java#L76) so that it maintains consistency.
Same for  [RPC](https://github.com/OpenNMS/opennms/blob/413fdc2b9451ba0c944141cd6b335abfcd8cd936/core/ipc/rpc/common/src/main/java/org/opennms/core/rpc/common/RpcStrategy.java#L76)

**Comment by jeffgdotorg** [comment]
Latest changes use the strategy method calls while still accounting for all three strategies. Please confirm suitability @cgorantla and thank you!

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 2]
```diff
@@ -131,10 +131,25 @@
     <th>Database Version:</th>
     <td><%=dbVersion%></td>
   </tr>
+<%
+    String globalMBStrategy, sinkMBStrategy, rpcMBStrategy;
+    globalMBStrategy = org.opennms.core.rpc.common.RpcStrategy.getRpcStrategy().getDescr();
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:136]: Think you need to import these packages.

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 3]
```diff
@@ -131,10 +131,25 @@
...
+    if (globalMBStrategy != null) {
+        sinkMBStrategy = globalMBStrategy;
+        rpcMBStrategy = globalMBStrategy;
+    } else {
+        sinkMBStrategy = org.opennms.core.ipc.sink.common.SinkStrategy.getSinkStrategy().getDescr();
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:141]: Sufficient to get Sink/RPC strategies here.

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 4]
```diff
@@ -131,10 +131,25 @@
...
+        rpcMBStrategy = org.opennms.core.rpc.common.RpcStrategy.getRpcStrategy().getDescr();
+    }
+%>
   <tr>
     <th>Time-Series Strategy:</th>
     <td><%=TimeSeries.getTimeseriesStrategy().getDescr()%></td>
   </tr>
+  <tr>
+    <th>IPC Strategy:</th>
+    <td>Sink: <%=sinkMBStrategy%>; RPC: <%=rpcMBStrategy%></td>
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:151]: Depending on how it looks on the web page, we should choose between `Descr` and `name`.
I prefer using `name`

File: opennms-webapp/src/main/webapp/about/index.jsp (hunk 2 above)
Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/about/index.jsp:136]: Importing doesn't seem to do the job; it still breaks on build and I get an additional message:

```
An error occurred at line: [21] in the generated java file: [/Users/jeffg/git/foundation-work/opennms-webapp/target/jsp-classes/org/apache/jsp/about/index_jsp.java]
Only a type can be imported. org.opennms.core.ipc.sink.common.SinkStrategy resolves to a package
```

Same result with either foundation-2019 or foundation-2020. Does the `opennms-webapp` POM perhaps need updating?

Comment (cgorantla) [reply opennms-webapp/src/main/webapp/about/index.jsp:136]: Guess you have to bring both dependencies into `opennms-webapp` and import  the packages [here](https://github.com/OpenNMS/opennms/blob/b0a7c12a3c3bf81f5886b1388b5488c918235e31/opennms-webapp/src/main/webapp/about/index.jsp#L35)

Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/about/index.jsp]: Got both dependencies added and ran it through full smoke, which passed. I think this should do the job.
Also changed from `descr` to `name` in the page.

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 5]
```diff
@@ -135,6 +137,21 @@
     <th>Time-Series Strategy:</th>
     <td><%=TimeSeries.getTimeseriesStrategy().getDescr()%></td>
   </tr>
+<%
+    String globalMBStrategy, sinkMBStrategy, rpcMBStrategy;
+    globalMBStrategy = org.opennms.core.rpc.common.RpcStrategy.getRpcStrategy().getName();
+    if (globalMBStrategy != null) {
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:143]: Didn't get why you still need `globalMBStrategy` here.

File: opennms-webapp/src/main/webapp/about/index.jsp [hunk 6]
```diff
@@ -135,6 +137,21 @@
...
+        sinkMBStrategy = globalMBStrategy;
+        rpcMBStrategy = globalMBStrategy;
+    } else {
+        sinkMBStrategy = org.opennms.core.ipc.sink.common.SinkStrategy.getSinkStrategy().getName();
```
Comment (cgorantla) [inline opennms-webapp/src/main/webapp/about/index.jsp:147]: you can omit package here.

File: core/web-assets/package-lock.json [hunk 7]
```diff
@@ -4330,9 +4330,9 @@
             }
         },
         "dot-prop": {
-            "version": "4.2.0",
-            "resolved": "https://registry.npmjs.org/dot-prop/-/dot-prop-4.2.0.tgz",
-            "integrity": "sha512-tUMXrxlExSW6U2EXiiKGSBVdYgtV8qlHL+C10TsW4PURY/ic+eaysnSkwB4kA/mBlCyy/IKDJ+Lc3wbWeaXtuQ==",
+            "version": "4.2.1",
```
Comment (cgorantla) [inline core/web-assets/package-lock.json:4333]: Guess this is unintentional change

**Comment by jeffgdotorg** [comment]
Screenshot of about page, using RPMs from full smoke pipeline (using `getDescr()`):
image

Same RPMs with about/index.jsp hand-modified to use `getName()`:
image

File: opennms-webapp/src/main/webapp/about/index.jsp (hunk 5 above)
Comment (cgorantla) [reply opennms-webapp/src/main/webapp/about/index.jsp:143]: Think you are trying to use `globalMBStrategy` in order to get both strategy names to be `jms`. Instead I think we can change `CAMEL_SINK_STRATEGY_NAME` to `jms`

**Comment by cgorantla** [comment]
> Screenshot of about page, using RPMs from full smoke pipeline (using `getDescr()`):
> image
> 
> Same RPMs with about/index.jsp hand-modified to use `getName()`:
> image

`name` looks fine to me.

File: opennms-webapp/src/main/webapp/about/index.jsp (hunk 5 above)
Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/about/index.jsp:143]: Guess I misunderstood earlier in our conversation. In this case, the scriptlet is totally unnecessary. Will change accordingly.

File: opennms-webapp/src/main/webapp/about/index.jsp (hunk 6 above)
Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/about/index.jsp:147]: Of course.

File: core/web-assets/package-lock.json (hunk 7 above)
Comment (jeffgdotorg) [reply core/web-assets/package-lock.json:4333]: > Guess this is unintentional change

I think is must have been. I certainly didn't make it intentionally.

Comment (RangerRick) [reply core/web-assets/package-lock.json:4333]: That probably came from a merge.  It's made its way through the various branches `foundation-2019` and up.

Comment (jeffgdotorg) [reply core/web-assets/package-lock.json]: I think you're right @RangerRick . It doesn't show up in a diff done today between the PR branch and `foundation-2019`.

**Review by cgorantla — APPROVED** [review]
Thanks for the updates Jeff. Looks good to me.

**Pull request merged**